ed25519.go

```diff
@@ -110,6 +110,16 @@ func Verify(publicKey *[PublicKeySize]byte, message []byte, sig *[SignatureSize]
 	if !A.FromBytes(publicKey) {
 		return false
 	}
+	return VerifyUncompressedKey(publicKey, &A, message, sig)
+}
+
+// VerifyUncompressedKey returns true iff sig is a valid signature of message by publicKey.
+// This takes in the uncompressed form of the public key (A) to avoid computing that internally.
+func VerifyUncompressedKey(publicKey *[PublicKeySize]byte, A *edwards25519.ExtendedGroupElement, message []byte, sig *[SignatureSize]byte) bool {
```

Comment:
see https://github.com/tendermint/ed25519/pull/7#discussion_r196539762

Reply:
Replied there too

Reply:
Also, as this interna become public now, it would be good to point to the section in the paper that deals with the compressed/uncompressed key. Such that a user who is expecting a API ala Sign/Verify can learn why this is different here.